### PR TITLE
Provide more context regarding the compat plugin + caveats

### PR DESCRIPTION
**Pull request by rinatkhaziev**
README updates regarding why we need this as well as some caveats and upgrade instructions

### CODE REVIEW

**Comment by rinatkhaziev** [comment]
@thekaitlin @brettshumaker @WPprodigy how does this read?

File: README.md [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+wpcom_vip_load_plugin( 'my-plugin' );
+```
+
+#### Versioned
+```
+// WP.com loads from `vip/plugins/my-plugin-2.0`
+wpcom_vip_load_plugin( 'my-plugin', 'plugins', '2.0' );
+
+// VIP Go loads from `plugins/`
+wpcom_vip_load_plugin( 'my-plugin' );
```
Comment (brettshumaker) [inline README.md]: Should this include a version to show how that gets handled on Go?

Comment (rinatkhaziev) [reply README.md]: Yes, yes it should :) Thanks for the catch!

**Comment by WPprodigy** [comment]
LGTM

**Pull request merged**